java-api/src/main/java/com/sap/cloud/security/token/Token.java

```diff
@@ -45,6 +48,13 @@ static Token create(String jwt) {
 		if (services.isEmpty()) {
 			throw new ProviderNotFoundException("No TokenFactory implementation found in the classpath");
 		}
+		if (services.size() > 1) {
+			LOGGER.error("More than 1 service provider found, will return first one from the list");
```

Comment:
does it mean, the error is also logged, in case one has specified a custom HttpClientFactory?
then you have two SPI, the DefaultHttpClientFactory.java  and the custom one...

Reply:
> Personally I dislike this solution as it makes the implementation too complicated... maybe we should strive for using SPIs only for custom enhancements...?

then the only way I see is to educate users to set in the META-INF.services file the factory classes in the right order -> first in the file is also first in the Java List

Reply:
the problem here is that SPI does not document the behaviour in case of multiple SPI files...